net/tcp/tcp_send_buffered.c

```diff
@@ -1172,7 +1177,24 @@ ssize_t psock_tcp_send(FAR struct socket *psock, FAR const void *buf,
               goto errout_with_lock;
             }
 
-          net_lockedwait_uninterruptible(&conn->snd_sem);
+          if (remain != UINT_MAX)
```

Comment:
this code is repeated 3 times. Maybe we can build a small support function and use it? Something like:
```
uint32_t tcp_send_gettimeout(uint32_t start, uint32_t remain)
{
  uint32_t elapse;

  if (remain != UINT_MAX)
    {
      elapse = TICK2MSEC(clock_systime_ticks()) - start;
      if (elapse >= remain)
        {
          remain = 0;
        }
      else
        {
          remain -= elapse;
        }
  }

  return remain;
}
```